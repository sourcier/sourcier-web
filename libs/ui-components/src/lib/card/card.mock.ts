export const href = '#';

export const heading = 'title';

export const copy =
  'Lorem ipsum dolor sit amet, enim officia aliquip ipsum ullamco irure eiusmod esse ullamco est nulla Lorem dolor nulla exercitation proident.';
