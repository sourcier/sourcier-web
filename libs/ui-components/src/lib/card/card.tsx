import { BsArrowRight } from 'react-icons/bs';

export interface CardProps {
  href: string;
  heading: string;
  copy: string;
  image?: React.ReactNode;
}

export function Card({ href, heading, copy, image }: CardProps) {
  return (
    <div className="overflow-hidden rounded-lg shadow-xl card bg-base-100">
      {image && image}
      <div className="card-body">
        <h2 className="card-title">{heading}</h2>
        <p className="line-clamp-3">{copy}</p>
        <div className="justify-center mt-4 card-actions">
          <a className="flex-1 btn btn-primary" href={href}>
            Read More <BsArrowRight className="ml-3" />
          </a>
        </div>
      </div>
    </div>
  );
}

export default Card;
