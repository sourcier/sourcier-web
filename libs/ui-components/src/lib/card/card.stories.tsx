import { Story, Meta } from '@storybook/react';

import { Card, CardProps } from './card';
import { href, heading, copy } from './card.mock';

export default {
  component: Card,
  title: 'Card',
} as Meta;

const Template: Story<CardProps> = (args) => <Card {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  href,
  heading,
  copy,
};
