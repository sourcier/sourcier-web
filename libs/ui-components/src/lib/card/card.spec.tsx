import renderer from 'react-test-renderer';

import Card from './card';
import { href, heading, copy, imgSrc, imgAlt } from './card.mock';

describe('Card', () => {
  it('should render successfully', () => {
    const tree = renderer
      .create(
        <Card
          href={href}
          heading={heading}
          copy={copy}
          imgSrc={imgSrc}
          imgAlt={imgAlt}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
