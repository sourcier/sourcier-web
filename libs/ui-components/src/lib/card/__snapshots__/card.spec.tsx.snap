// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render successfully 1`] = `
<div
  className="shadow-xl card w-96 bg-base-100"
>
  <div
    className="card-body"
  >
    <h2
      className="card-title"
    >
      title
    </h2>
    <p
      className="line-clamp-3"
    >
      Lorem ipsum dolor sit amet, enim officia aliquip ipsum ullamco irure eiusmod esse ullamco est nulla Lorem dolor nulla exercitation proident.
    </p>
    <div
      className="justify-center mt-4 card-actions"
    >
      <a
        className="flex-1 btn btn-primary"
        href="#"
      >
        Read More 
        <svg
          className="ml-3"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
            fillRule="evenodd"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
